src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -234,22 +234,23 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
                                 : _diagnosticKind == DiagnosticKind.AnalyzerSemantic;
                         }
 
+                        var lazyActiveFileState = new Lazy<ActiveFileState>(() => stateSet.GetOrCreateActiveFileState(_document.Id));
```

Comment:
>var lazyActiveFileState = new Lazy<ActiveFileState>(() => stateSet.GetOrCreateActiveFileState(_document.Id)); [](http://example.com/codeflow?start=24&length=109)

Doesn't this do a couple allocations per loop? I wonder if it would be nicer if the SupportAnalysisKind check was moved from TryAddCachedDocumentDiagnosticsAsync out to here and then getting rid of the lazy. Something like:

includeSyntax = includeSyntax && analyzer.SupportAnalysisKind(AnalysisKind.Syntax);
includeSemantic = includeSemantic && _document is Document && analyzer.SupportAnalysisKind(Semantic.Syntax);

if (includeSyntax || includeSemantic)
{
  var activeFileState = stateSet.GetOrCreateActiveFileState(_document.Id);

   if (includeSyntax)
   {
...
   }

   if (includeSemantic)
   {
...
   }
}

Reply:
Any reason why you didn't do the SupportsAnalysisKind checks earlier like in my code snippet? I liked that approach because it prevented the GetOrCreateActiveFileState call when the SupportsAnalysisKind methods return false.

Reply:
Done now, thanks!